### PR TITLE
NMS-15870: Elastic config updates

### PR DESCRIPTION
**Pull request by mmahacek**
Expand Elastic config to allow per-server credentials, multiple servers, and alarm history options

Jira: https://opennms.atlassian.net/browse/NMS-15870

### CODE REVIEW

**Comment by mmahacek** [comment]
@mershad-manesh I'm trying to tackle some updates to the elastic config, and wanted some input to see if I'm going in the right direction.   This changes the way that the elastic config is handled to allow:

* multiple elastic hosts
* unique credentials per host
* enablement of alarm history feature
* configuration of any arbitrary elastic config setting

I know there are some issues to finish working out, including handling passwords that have an `@` in them and verifying the sentinel config still works.

File: horizon/Chart.yaml [hunk 1]
```diff
@@ -17,13 +17,13 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 1.0.0
+version: 1.0.1
```
Comment (mershad-manesh) [inline horizon/Chart.yaml:20]: We shouldn't increment the chart versions unless we are ready to publish all changes in the main branch.

File: minion/Chart.yaml [hunk 2]
```diff
@@ -17,13 +17,13 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 1.0.0
+version: 1.0.1
```
Comment (mershad-manesh) [inline minion/Chart.yaml:20]: We shouldn't increment the chart versions unless we are ready to publish all changes in the main branch.

Comment (mmahacek) [reply minion/Chart.yaml:20]: I know, but the tests were failing when there wasn't a version bump.

File: horizon/scripts/onms-elastic-init.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+#
+# Intended to be used as part of an InitContainer expecting to run to configure Elastic settings
+#
+# External environment variables used by this script:
```
Comment (mershad-manesh) [inline horizon/scripts/onms-elastic-init.sh:5]: Can you add a list of variables used in this script.

FYI, there is a limit on config map size (by default it is 1mb).

File: horizon/scripts/onms-sentinel-init.sh [hunk 4]
```diff
@@ -48,13 +45,10 @@ OPENNMS_DATABASE_CONNECTION_MAXPOOL=${OPENNMS_DATABASE_CONNECTION_MAXPOOL-50}
...
 KAFKA_SECURITY_PROTOCOL=${KAFKA_SECURITY_PROTOCOL-SASL_PLAINTEXT}
-ELASTICSEARCH_INDEX_STRATEGY_FLOWS=${ELASTICSEARCH_INDEX_STRATEGY_FLOWS-daily}
-ELASTICSEARCH_REPLICATION_FACTOR=${ELASTICSEARCH_REPLICATION_FACTOR-2}
-ELASTICSEARCH_NUM_SHARDS=${ELASTICSEARCH_NUM_SHARDS-6}
 
 # Wait for Dependencies
-if [[ -v ELASTICSEARCH_SERVER ]]; then
-  wait_for ${ELASTICSEARCH_SERVER}
+if [[ -v ELASTICSEARCH_HOSTS ]]; then
+  wait_for ${ELASTICSEARCH_HOSTS}
```
Comment (mershad-manesh) [inline horizon/scripts/onms-sentinel-init.sh:51]: I'm not sure how well the `wait_for` function works for multiple hosts; we may need to improve it.

File: minion/Chart.yaml (hunk 2 above)
Comment (mershad-manesh) [reply minion/Chart.yaml:20]: Yup, that is ok; That is expected (for now)

File: horizon/scripts/onms-sentinel-init.sh (hunk 4 above)
Comment (mmahacek) [reply horizon/scripts/onms-sentinel-init.sh:51]: I know this is broken and I'm looking at what I can do to handle that.

**Pull request closed without merging**